### PR TITLE
Implement swagger spec

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Added initial swagger spec based on the documentation.

### How to review

Check the swagger spec

### Who can review

Anyone with swagger 🕺 😎

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+          description: "The hierarchy was found and returned"
+          schema:
+            $ref: '#/definitions/HierarchyResponse'
+        400:
+          $ref: '#/responses/InstanceNotFound'
+        404:
+          $ref: '#/responses/DimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /hierarchies/{instance_id}/{dimension_name}}/{code_id}:
```
Comment (eldeal) [inline swagger.yaml]: Unnecessary extra } after dimension_name

File: swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        description: "A link to the parent node of this node"
+        example: "http://localhost:8080/hierarchies/i1234/CPIAggregate/09"
+        type: string
+  CodeLink:
+    type: object
+    readOnly: true
+    properties:
+      href:
+        description: "A link to the code related to this node"
+        example: "http://localhost:8080//code-lists/c345/codes/09.1"
```
Comment (eldeal) [inline swagger.yaml]: Extra / before code-lists

File: swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      has_data:
+        type: boolean
+        description: "True if the instance has an observation for this code"
+      children:
+        type: array
+        description: "The child nodes of this node in the hierarchy"
+        items:
+          type: object
+          items:
+            $ref: '#/definitions/CodeChild'
```
Comment (eldeal) [inline swagger.yaml]: This is not coming through on the example response for the endpoint - feels like it should be but not sure why - I might be missing something that explains why it actually shouldnt.

File: swagger.yaml (hunk 1 above)
Comment (CarlHembrough) [reply swagger.yaml]: Done

File: swagger.yaml (hunk 2 above)
Comment (CarlHembrough) [reply swagger.yaml]: Done

File: swagger.yaml (hunk 3 above)
Comment (CarlHembrough) [reply swagger.yaml]: I have added some example data for child nodes

**Pull request merged**